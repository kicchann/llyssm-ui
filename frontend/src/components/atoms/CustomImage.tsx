import { styled } from '@mui/material';
import React from 'react';

interface CustomImageProps {
  src: string;
  alt: string;
  variant?: 'marker' | 'layer';
}

const StyledCustomImage = styled('img')<CustomImageProps>`
  display: block;
  margin-left: auto;
  margin-right: auto;
  width: 100%;

  /* variantが 'layer' の場合の追加スタイル */
  ${({ variant }) =>
    variant === 'layer' &&
    `
    height: 100%;
    object-fit: contain;
    cursor: grab;
  `}
`;

export const CustomImage: React.FC<CustomImageProps> = ({
  src,
  alt,
  variant = 'marker',
}) => {
  return <StyledCustomImage src={src} alt={alt} variant={variant} />;
};
