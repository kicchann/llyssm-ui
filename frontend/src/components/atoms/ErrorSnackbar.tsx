import { Alert, Snackbar } from '@mui/material';
import React from 'react';

// TODO: このコンポーネントは、エラーメッセージを表示するSnackbarです。
// TODO: あとで実装します。

interface ErrorSnackbarProps {
  open: boolean;
  message: string;
  onClose: () => void;
}

export const ErrorSnackbar: React.FC<ErrorSnackbarProps> = ({
  open,
  message,
  onClose,
}) => {
  return (
    <Snackbar open={open} autoHideDuration={6000} onClose={onClose}>
      <Alert onClose={onClose} severity="error">
        {message}
      </Alert>
    </Snackbar>
  );
};
