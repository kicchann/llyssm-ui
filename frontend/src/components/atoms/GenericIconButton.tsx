import {
  Close,
  GridView,
  KeyboardDoubleArrowLeft,
  KeyboardDoubleArrowRight,
  Layers,
  MoreVert,
} from '@mui/icons-material';
import { IconButton, styled } from '@mui/material';

interface GenericIconButtonProps {
  title: string;
  onClick?: () => void;
  iconType?:
    | 'closeSidebar'
    | 'openSidebar'
    | 'openDrawer'
    | 'closeDrawer'
    | 'layerTileGrid'
    | 'settingMenu'
    | 'close';
  dataTestId?: string;
}

const icons = {
  closeSidebar: <KeyboardDoubleArrowLeft />,
  openSidebar: <KeyboardDoubleArrowRight />,
  openDrawer: <Layers />,
  closeDrawer: <Layers />,
  layerTileGrid: <GridView />,
  settingMenu: <MoreVert />,
  close: <Close />,
};

const StyledIconButton = styled(IconButton)`
  background-color: rgba(255, 255, 255, 0.5);
`;

export const GenericIconButton: React.FC<GenericIconButtonProps> = ({
  title,
  onClick,
  iconType = 'close',
  dataTestId = 'generic-icon-button',
}) => {
  console.log(dataTestId);
  return (
    <StyledIconButton
      title={title}
      onClick={onClick ? onClick : () => {}}
      data-testid={dataTestId}
      aria-label={title}
    >
      {icons[iconType]}
    </StyledIconButton>
  );
};
