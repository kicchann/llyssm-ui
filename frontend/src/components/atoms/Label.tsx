// src/components/atoms/Label.tsx
import { Typography } from '@mui/material';
import React from 'react';

interface LabelProps {
  text: string;
  color?: string;
}

export const Label: React.FC<LabelProps> = ({ text, color = 'black' }) => (
  <Typography style={{ color }}>{text}</Typography>
);
