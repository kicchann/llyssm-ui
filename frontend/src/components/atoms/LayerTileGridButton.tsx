import { GridView } from '@mui/icons-material';
import { IconButton, styled } from '@mui/material';
import React from 'react';

interface LayerTileGridButtonProps {
  onClick?: () => void;
}

const StyledIconButton = styled(IconButton)`
  background-color: rgba(255, 255, 255, 0.5);
`;

export const LayerTileGridButton: React.FC<LayerTileGridButtonProps> = ({
  onClick,
}) => {
  return (
    <StyledIconButton onClick={onClick} title="レイヤー一覧">
      <GridView />
    </StyledIconButton>
  );
};
