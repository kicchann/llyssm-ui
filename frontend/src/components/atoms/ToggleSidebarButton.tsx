import {
  KeyboardDoubleArrowLeft,
  KeyboardDoubleArrowRight,
} from '@mui/icons-material';
import { IconButton, styled } from '@mui/material';
import React from 'react';

interface ToggleSidebarButtonProps {
  isSidebarOpen: boolean;
  onToggle: () => void;
}

const StyledIconButton = styled(IconButton)`
  background-color: rgba(255, 255, 255, 0.5);
`;

export const ToggleSidebarButton: React.FC<ToggleSidebarButtonProps> = ({
  isSidebarOpen,
  onToggle,
}) => {
  return (
    <StyledIconButton
      onClick={onToggle}
      title={isSidebarOpen ? 'サイドバーを閉じる' : 'サイドバーを開く'}
    >
      {isSidebarOpen ? (
        <KeyboardDoubleArrowLeft />
      ) : (
        <KeyboardDoubleArrowRight />
      )}
    </StyledIconButton>
  );
};
