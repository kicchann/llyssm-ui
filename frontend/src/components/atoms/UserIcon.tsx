import { AccountCircle } from '@mui/icons-material';
import React from 'react';

interface UserIconProps {
  size?: number;
  color?: string;
}

export const UserIcon: React.FC<UserIconProps> = ({
  size = 16, //24,
  color = 'inherit',
}) => <AccountCircle style={{ fontSize: size, color: color }} />;
