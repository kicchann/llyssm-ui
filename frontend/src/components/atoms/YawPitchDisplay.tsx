import { Box, Typography } from '@mui/material';
import React from 'react';

interface YawPitchDisplayProps {
  yaw: number;
  pitch: number;
}

export const YawPitchDisplay: React.FC<YawPitchDisplayProps> = ({
  yaw,
  pitch,
}) => {
  return (
    <Box
      sx={{
        position: 'absolute',
        bottom: 45,
        right: 10,
        backgroundColor: 'rgba(0, 0, 0, 0.5)',
        padding: '5px 10px',
        borderRadius: '5px',
      }}
    >
      <Typography variant="body2" color="white">
        Yaw: {yaw.toFixed(2)}°
      </Typography>
      <Typography variant="body2" color="white">
        Pitch: {pitch.toFixed(2)}°
      </Typography>
    </Box>
  );
};
