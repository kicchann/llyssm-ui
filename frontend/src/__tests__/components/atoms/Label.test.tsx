import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Label } from '../../../components/atoms/Label';

describe('Label component', () => {
  it('renders text correctly', () => {
    // テスト用のプロパティ
    const labelText = 'Test Label';

    // コンポーネントをレンダリング
    render(<Label text={labelText} />);

    // テキストが正しく表示されているか確認
    const renderedLabel = screen.getByText(labelText);
    expect(renderedLabel).toBeInTheDocument();
  });

  it('renders text with color correctly', () => {
    const labelText = 'Colored Label';
    const labelColor = 'red';

    // 色指定付きでコンポーネントをレンダリング
    render(<Label text={labelText} color={labelColor} />);

    // テキストが正しく表示されているか確認
    const renderedLabel = screen.getByText(labelText);

    // スタイルのcolorが正しいか確認
    expect(renderedLabel).toHaveStyle({ color: labelColor });
  });
});

test('Label component snapshot', () => {
  const { asFragment } = render(<Label text="Snapshot Test" color="red" />);
  expect(asFragment()).toMatchSnapshot();
});
