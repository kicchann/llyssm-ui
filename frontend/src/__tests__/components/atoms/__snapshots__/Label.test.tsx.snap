// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component snapshot 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
    style="color: red;"
  >
    Snapshot Test
  </p>
</DocumentFragment>
`;
